docs/react-testing-library/setup.mdx

```diff
@@ -491,3 +491,137 @@ And register it using mocha's `-r` flag:
 ```
 mocha -r ./mocha-watch-cleanup-after-each.js
 ```
+
+### Vitest
+
+To test web applications with [Vitest](https://vitest.dev/),
+you need to install a browser-like environment such as `jsdom`.
+
+```bash npm2yarn
+npm install --save-dev vitest jsdom
+npm install --save-dev @testing-library/react @testing-library/jest-dom
+```
+
+In the `package.json`, you can setup the script for running the test by Vitest.
+
+```json
+// package.json
+{
+  "scripts": {
+    "test": "vitest",
+    "coverage": "vitest run --coverage"
+  }
+}
+```
+
+Then set the `environment` to `jsdom` in the `vite.config.js`.
+
+```js
+// vite.config.js
+export default defineConfig({
+  test: {
+    // highlight-next-line
+    environment: "jsdom",
+  },
+})
+```
+
+If you are using TypeScript, add a triple slash command at the top of the
+config file to reference Vitest types.
+
+```ts
+// vite.config.ts
+// highlight-next-line
+/// <reference types="vitest" />
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+  test: {
+    environment: 'jsdom',
+  },
+});
+```
+
+> **NOTE**
+>
+> An alternative to JSDOM is [Happy DOM](https://github.com/capricorn86/happy-dom).
+> You can use either `jsdom` or `happy-dom` to create a browser-like environment.
+
+By default, Vitest does not provide [global](https://vitest.dev/config/#globals) APIs.
+To use them without importing them like jest,
+you can set `globals` to `true` in the `vite.config.js`.
+
+```js
+// vite.config.js
+export default defineConfig({
+  test: {
+    environment: "jsdom",
+    // highlight-next-line
+    globals: true,
+  },
+})
+```
+
+If you are using TypeScript, you need to add `vitest/global` to the `types` field in
+your `tsconfig.json`.
+
+```json
+// tsconfig.json
+{
+  "compilerOptions": {
+    "types": ["vitest/globals"]
+  }
+}
+```
+
+> **NOTE**
+>
+> Setting `test.globals` to `true` also improves compatibility when using
+> testing-library with Vitest, such as extending matchers and auto-cleanup, so
+> you don't have to set them up manually.
+
+To extends the matchers in Vitest, you just need to import the `@testing-library/jest-dom`
+after setting `test.globals` to `true`. To do this, create a `setupTests` file
+and import `@testing-library/jest-dom`.
+
+```js
+// src/setupTests.js
+import '@testing-library/jest-dom'
+```
+
+Vitest will run `src/setupTests.js` before each test file, as configured in `setupFiles`.
+
+```js
+// vite.config.js
+export default defineConfig({
+  test: {
+    environment: "jsdom",
+    globals: true,
+    // highlight-next-line
+    setupFiles: "src/setupTests.js"
+  },
+})
+```
```

Comment:
@eps1lon do you suggest removing this paragraph from this PR?

Reply:
Yep. jest-dom docs should handle this.

Reply:
I think for Vitest you'd use [`vitest-dom`](https://github.com/chaance/vitest-dom) instead of `jest-dom`

Reply:
I don't think many people use `vitest-dom` for testing react components with Vitest. Additionally, without addressing the step of creating the `setupTests` file, it can be difficult to set up the testing environment for React. Please prove me wrong.

Reply:
That's not really relevant. `jest-dom` is not for usage with Vitest. `vitest-dom` is. I don't see how the popularity of `vitest-dom` is relevant here?

Reply:
Thanks for your reply. I understand your points now. My documentation appears to be written for `@testing-library/jest-dom` but not for `@testing-library/react`, right?

Should I close this Pull Request here and create a new one for `@testing-library/jest-dom`? Since you mentioned that `jest-dom` is not meant to be used with Vitest, I'm considering whether I should create a Pull Request. I know that Vitest is compatible with Jest. Does this mean that Vitest is also compatible with jest-dom? What do you think?

Reply:
Your PR is targetting `@testing-library/react`. There's no need to target `@testing-library/jest-dom` since that library is not supposed to be used with Vitest.

Reply:
@eps1lon thanks. I've removed the paragraph. Could you check it again?